### DISCUSSION TITLE
[Firebase Cloud Messaging] iOS devices not receiving notifications, test notifications work

### DISCUSSION THREAD

**Discussion by dancamdev** [firebase/flutterfire#5112] [Q&A]
I've recently implemented the Firebase cloud messaging on my app and I'm having issues with the delivery only on iOS devices.

on Android I get:

6.300 sent -> 2.029 received -> 2.028 impression -> 218 open

on the other hand, if I take the iOS app, I got:

12.024 sent -> 0 received -> 0 impression -> 2 opens

The opens on iOS make no sense of course, but I wonder why people are not receiving the notifications. The implementation is correct, I've checked it multiple times, and test notifications sent via the Firebase Console are working flawlessly.

What follows are the screenshots of the data I'm was talking above.

<img width="951" alt="Screenshot 2021-02-23 at 11 51 53" src="https://user-images.githubusercontent.com/43640732/108834348-5eda2700-75ce-11eb-917e-cac73004f3cf.png">
<img width="956" alt="Screenshot 2021-02-23 at 11 51 36" src="https://user-images.githubusercontent.com/43640732/108834355-600b5400-75ce-11eb-887b-b88653e82b27.png">

**Comment by Sabatin** [comment]
Same problem. I don't know how to fix it...

**Comment by SkylarMark** [comment]
Did notifications work with TestFlight?

@dancamdev please check your certificate file. It may be an issue with the certificate. You can find it inside Project Settings > Cloud Messaging.

Try replacing the certificate.

![image](https://user-images.githubusercontent.com/58359732/108905478-c4b5c580-7645-11eb-83fc-fcd95bfe2d7f.png)

Production is the one you are looking for.

and make sure that a certificate is installed in Apple Identifier.

![image](https://user-images.githubusercontent.com/58359732/108906110-88369980-7646-11eb-878b-e5e851023f4a.png)

I hope this solves your problem.

**Reply by dancamdev** [reply]
I haven't tried on TestFlight, I'll try shipping an update with the updated certificate and let you know 🙌 thanks a lot for the help so far ☝️
